src/emmetHelper.ts

```diff
@@ -23,8 +23,11 @@ import { parseSnippets, SnippetsMap, syntaxes } from './configCompat';
 export { FileService, FileType, FileStat };
 
 let l10n: { t: (message: string) => string };
+let getProfileOfCurrentLanguageId = (profilesConfig: any) => null;
 try {
-	l10n = require('vscode').l10n;
+	var vscode = require('vscode');
+	l10n = vscode.l10n;
+	getProfileOfCurrentLanguageId = (profilesConfig: any) => profilesConfig[vscode.window.activeTextEditor.document.languageId];
```

Comment:
Wouldn't this section only run right when the package is loaded?
To override the use of `profilesConfig[syntax]`, you might have to change the `getProfile` API or provide a new one.